app/services/recurring_contribution/subscriptions/update_status.rb

```diff
@@ -0,0 +1,21 @@
+class RecurringContribution::Subscriptions::UpdateStatus
+  def initialize(subscription_id, current_status)
```

Comment:
Why don't you rename `subscription_id` for `subscription_code`? Since you do the `find_by(subscription_code:....)`